frontend/src/libs/components/form/libs/helpers/handle-server-error.helper.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  type FieldPath,
+  type FieldValues,
+  type UseFormSetError,
+} from 'react-hook-form';
+
+import { ServerErrorType } from '~/libs/enums/enums.js';
+import { type ServerSerializedError } from '~/libs/packages/store/store.js';
+import {
+  type FormField,
+  type ServerCommonErrorResponse,
+  type ServerValidationErrorResponse,
+} from '~/libs/types/types.js';
+
+import { SERVER_ERROR_SYMBOL } from '../consts/consts.js';
+
+const handleServerError = <T extends FieldValues>(
+  error: ServerSerializedError,
+  setError: UseFormSetError<T>,
+  fields: FormField<T>[],
+): void => {
+  switch (error.errorType) {
+    case ServerErrorType.COMMON: {
+      assignCommonErrors(fields, error, setError);
+      break;
+    }
+    case ServerErrorType.VALIDATION: {
+      assignValidationErrors(fields, error, setError);
+    }
+  }
+};
+
+function assignCommonErrors<T extends FieldValues>(
+  fields: FormField<T>[],
+  commonError: Partial<ServerCommonErrorResponse>,
+  setError: UseFormSetError<T>,
+): void {
+  for (const field of fields) {
+    if (field.associateServerErrors) {
+      for (const errorDescriptor of field.associateServerErrors) {
+        if (
+          typeof errorDescriptor === 'object' &&
+          errorDescriptor.errorMessage === commonError.message
+        ) {
+          setError(field.name, errorDescriptor.error, errorDescriptor.options);
+        } else if (errorDescriptor === commonError.message) {
+          setError(field.name, {
+            type: SERVER_ERROR_SYMBOL,
+            message: errorDescriptor,
+          });
+        }
+      }
+    }
+  }
+}
+
+function assignValidationErrors<T extends FieldValues>(
```

Comment:
Try to follow single code style here we have mix of arrow function `const handleServerError = () =>` and function declaration `function assignValidationErrors`